src/base/server_type.cpp

```diff
@@ -201,6 +208,73 @@ void setup_tran_server_params_on_single_node_config ()
   prm_set_bool_value (PRM_ID_REMOTE_STORAGE, true);
 }
 
+int setup_tran_server_params_on_ha_mode ()
+{
+  assert (!HA_DISABLED ());
+
+  char *page_hosts_new_value = NULL;
+  constexpr size_t MAX_BUFSIZE = 4096;
+  int new_value_size = 0;
+
+  char ha_node_list[MAX_BUFSIZE];
+  char *str, *savep;
+
+  int port_id = prm_get_master_port_id ();
+
+  page_hosts_new_value = (char *) calloc (MAX_BUFSIZE, sizeof (char)); // free is called by sysprm_final()
+  if (page_hosts_new_value == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, MAX_BUFSIZE);
+      return ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+  new_value_size = MAX_BUFSIZE;
+
+  strncpy_bufsize (ha_node_list, prm_get_string_value (PRM_ID_HA_NODE_LIST));
```

Comment:
Q. What if `PRM_ID_HA_NODE_LIST` is not set and `ha_mode` is enabled?

Reply:
the default value of it, `prm_ha_node_list_default`, is NULL, but it's set to `localhost` somewhere?

Reply:
In `prm_tune_parameters ()`, if `ha_mode` is set to true, then it set the value of `ha_node_list` to `localhost@localhost`.
`prm_tune_parameters ()` is called in `sysprm_load_and_init ()`, and `sysprm_load_and_init ()` is the function that load the system parameters.